can't make "server" on Mac

I followed all instructions in the README, and on the `make server` step, I get
```
one-C02VV5SWHTD6:server bcohen$ make server
make: *** No rule to make target `server'.  Stop.
None-C02VV5SWHTD6:server bcohen$ 
```

You were correct, I was not at the top level,  But now I get the following:
```
Generating server key...
[ -d server/conf] || mkdir -p server/conf
/bin/sh: line 0: [: missing `]'
openssl genpkey -out server/conf/server.key -outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:4096
.....++
...++
>>> Generating server X.509 certificate.
>>> Enter the server's full hostname as the 'Common Name' (e.g. cuekeeper.mynet).
>>> Everything else can be left blank.

You are about to be asked to enter information that will be incorporated
into your certificate request.
What you are about to enter is what is called a Distinguished Name or a DN.
There are quite a few fields but you can leave some blank
For some fields there will be a default value,
If you enter '.', the field will be left blank.
-----
Country Name (2 letter code) []:
State or Province Name (full name) []:
Locality Name (eg, city) []:
Organization Name (eg, company) []:
Organizational Unit Name (eg, section) []:
Common Name (eg, fully qualified host name) []:localhost
Email Address []:
rm -rf _build/static
mkdir _build/static
mkdir: _build: No such file or directory
make: *** [server] Error 1
```

I continue the fight 😁.  My guess is this is macOS specifics, and folks on Linux don't have to do this (otherwise, it would be in the README).  I am on the latest Catalina build.
Anyway,  I was made to install **A LOT** of `opam` packages:

- `opam install js_of_ocaml-compiler`
- `opam install cohttp cohttp-lwt cohttp-lwt-jsoo irmin`
- `opam install irmin-indexeddb js_of_ocaml-tyxml omd react`
- `opam install sexplib tar uuidm`

Then, I was told to run:

- `dune external-lib-deps --missing --profile release ./js/client.bc.js`

Then I finally tried `make server`, and now am presented with the following:
```
None-C02VV5SWHTD6:cuekeeper-master bcohen$ make server
rm -rf _build/static
mkdir -p _build/static
cp -r resources _build/static/
dune build --profile=release ./js/client.bc.js
File "lib/dune", line 15, characters 7-19:
15 |   (run ocaml-crunch init -o ck_init.ml -m plain)))
            ^^^^^^^^^^^^
Error: Program ocaml-crunch not found in the tree or in PATH
 (context: default)
File "lib/ck_utils.ml", line 70, characters 2-30:
70 |   ReactiveData.RList.make_from init changes
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: Unbound value ReactiveData.RList.make_from
make: *** [server] Error 1
None-C02VV5SWHTD6:cuekeeper-master bcohen
```

Created the client successfully.

But for the server, I think we are almost there:
```
make server
rm -rf _build/static
mkdir -p _build/static
cp -r resources _build/static/
dune build --profile=release ./js/client.bc.js
cp _build/default/js/client.bc.js _build/static/resources/js/cuekeeper.js
sed 's!_build/default/js/client.bc.js!resources/js/cuekeeper.js!;s!var ck_use_server=false;!var ck_use_server=true;!' test.html > _build/static/index.html
ocaml-crunch _build/static -e html -e js -e css -e ico -o server/static.ml -m plain
Generating server/static.ml
Skipping generation of .mli
(cd server && mirage configure -t unix && make depend && make)
/bin/sh: mirage: command not found
make: *** [server] Error 127
```

--- a/Makefile
+++ b/Makefile
@@ -51,7 +51,7 @@ server/conf/server.pem: server/conf/server.key
 
 server: server/conf/server.pem
 	rm -rf _build/static
-	mkdir _build/static
+	mkdir -p _build/static
 	cp -r resources _build/static/
 	dune build --profile=release ./js/client.bc.js
 	cp _build/default/js/client.bc.js _build/static/resources/js/cuekeeper.js

--- a/server/config.ml
+++ b/server/config.ml
@@ -14,7 +14,7 @@ let packages = [
   package "irmin" ~min:"1.3.0" ~max:"2.0.0";
   package "irmin-git" ~min:"1.3.0" ~max:"2.0.0";
   package "irmin-mem" ~min:"1.3.0" ~max:"2.0.0";
-  package "tls";
+  package "tls" ~max:"0.11.0";
   package "cohttp-mirage";
 ]
 

